### PR TITLE
Update add_plugin.md

### PR DESCRIPTION
**Pull request by justinegeffen**
Updates to align with: https://github.com/mattermost/mattermost-developer-documentation/pull/747.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/add_plugin.md [hunk 1]
```diff
@@ -18,36 +18,34 @@ A brief description what your plugin does. Consider including screenshots to hel
...
 - [ ] The plugin provides detailed usage documentation with at least one screenshot of the plugin in action, list of features, and a development guide. This is typically a `README` file or a landing page on the web. The link to the documentation is set as `homepage_url` in the manifest. A great example is the [`README` of the GitHub plugin](https://github.com/mattermost/mattermost-plugin-github/blob/master/README.md).
 - [ ] For the current release and upcoming ones a changelog has to be published, with a link recorded in the `release_notes_url` property of the `plugin.json` manifest.
 - [ ] The plugin has to be out of Beta and be released with at least v1.0.0.
-- [ ] All configuration is accessible via the Mattermost UI.
-- [ ] The plugin ID defined in the manifest must not collide with the ID of an existing plugin in the Marketplace. It should follow [the documentation's suggested naming convention](https://developers.mattermost.com/extend/plugins/manifest-reference/#id).
+- [ ] All configuration is accessible via the Mattermost interface.
```
Comment (hanzei) [inline .github/ISSUE_TEMPLATE/add_plugin.md:43]: This isn't the same wording as in https://github.com/mattermost/mattermost-developer-documentation/pull/747/files

File: .github/ISSUE_TEMPLATE/add_plugin.md [hunk 2]
```diff
@@ -18,36 +18,34 @@ A brief description what your plugin does. Consider including screenshots to hel
...
+- [ ] The plugin ID defined in the manifest must not collide with the ID of an existing plugin in the Plugin Marketplace. It should follow [the documentation's suggested naming convention](https://developers.mattermost.com/extend/plugins/manifest-reference/#id).
```
Comment (hanzei) [inline .github/ISSUE_TEMPLATE/add_plugin.md:44]: Actually, the term used in the product is "Marketplace". https://github.com/mattermost/mattermost-developer-documentation/pull/747/ needs to get updated.

File: .github/ISSUE_TEMPLATE/add_plugin.md [hunk 3]
```diff
@@ -18,36 +18,34 @@ A brief description what your plugin does. Consider including screenshots to hel
...
+Please link to an open source repository and release that should be used for review. As Mattermost code reviews and builds all plugins itself when listing in the Marketplace, the link cannot point at an already-built plugin.
 -->
 
 ## Checklist
 <!--
 Go through this checklist and confirm every item.
 
-It's fine if your plugin doesn't fulfill every item (e.g. it isn't production ready yet). You can still submit it!
-You can also still do code changes while the plugin is in review and fix issues on the fly.
+It's fine if your plugin doesn't fulfill every item (e.g. it isn't production ready yet). You can still submit it! You can also still do code changes while the plugin is in review and fix issues on the fly.
```
Comment (hanzei) [inline .github/ISSUE_TEMPLATE/add_plugin.md:28]: The GitHub issue box is quite small. The line break was on purpose.

File: .github/ISSUE_TEMPLATE/add_plugin.md [hunk 4]
```diff
@@ -18,36 +18,34 @@ A brief description what your plugin does. Consider including screenshots to hel
...
 
-Even if your plugin isn't production-ready, it might still be added to the Marketplace as "Beta".
-See https://developers.mattermost.com/extend/plugins/community-plugin-marketplace/#beta-plugins for more details.
+Even if your plugin isn't production-ready, it might still be added to the Marketplace as "Beta". See https://developers.mattermost.com/extend/plugins/community-plugin-marketplace/#beta-plugins for more details.
```
Comment (hanzei) [inline .github/ISSUE_TEMPLATE/add_plugin.md:30]: Same her

**Review by hanzei — CHANGES_REQUESTED** [review]
Thanks for keeping the docs in sync :+1:

**Comment by justinegeffen** [comment]
@hanzei, this file is now aligned with: https://github.com/mattermost/mattermost-developer-documentation/pull/747.

File: .github/ISSUE_TEMPLATE/add_plugin.md [hunk 5]
```diff
@@ -18,36 +18,30 @@ A brief description what your plugin does. Consider including screenshots to hel
...
 Go through this checklist and confirm every item.
-
-It's fine if your plugin doesn't fulfill every item (e.g. it isn't production ready yet). You can still submit it!
-You can also still do code changes while the plugin is in review and fix issues on the fly.
-
-Even if your plugin isn't production-ready, it might still be added to the Marketplace as "Beta".
-See https://developers.mattermost.com/extend/plugins/community-plugin-marketplace/#beta-plugins for more details.
-
-If your plugin isn't production ready, please leave a comment stating whether you plan to fulfill the whole checklist or submit a "Beta" plugin.
+It's fine if your plugin doesn't fulfill every item (e.g. it isn't production ready yet). You can still submit it! You can also still do code changes while the plugin is in review and fix issues on the fly.
```
Comment (hanzei) [inline .github/ISSUE_TEMPLATE/add_plugin.md:27]: The line breakes were placed intentionally here they ensure the text is readable in the "Create Issue" text box without scrolling. Could you please revert the changes?

File: .github/ISSUE_TEMPLATE/add_plugin.md [hunk 6]
```diff
@@ -59,3 +53,15 @@ If your plugin isn't production ready, please leave a comment stating whether yo
...
+
+- [ ] The plugin must include detailed usage documentation with at least one screenshot of the plugin in action, list of features, and a development guide. This is typically a `README` file or a landing page on the web. The link to the documentation is set as `homepage_url` in the manifest. A great example is the [`README` of the GitHub plugin](https://github.com/mattermost/mattermost-plugin-github/blob/master/README.md). Typical components of documentation include:
+
+    * Requirements/Prerequisites
+    * Installation steps
+    * Configuration steps
+    * Usage
+    * Troubleshooting
+    * Screenshots (if available)
+    * Link or email address for help/support
```
Comment (hanzei) [inline .github/ISSUE_TEMPLATE/add_plugin.md:67]: Should the line from https://github.com/mattermost/mattermost-developer-documentation/pull/929/files#diff-01cbb01e3700f3a6f551852a98b49ab01ddff4600efe6642c6f813f803bfde3aR52 also get copied over?

**Review by hanzei — CHANGES_REQUESTED** [review]

**Review by hanzei — COMMENTED** [review]
Given that https://github.com/mattermost/mattermost-developer-documentation/pull/747 was closed, is this PR still valid? The docs in the repo and the dev docs should be in sync.

**Review by hanzei — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/add_plugin.md [hunk 7]
```diff
@@ -59,3 +56,17 @@ If your plugin isn't production ready, please leave a comment stating whether yo
 
 - [ ] The plugin must set a `min_server_version` in the manifest.
 - [ ] The plugin must work on all Mattermost versions greater than or equal to the `min_server_version`.
+
+**Documentation requirements (checked by a Technical Writer)**
+
+- [ ] The plugin must include detailed usage documentation with at least one screenshot of the plugin in action, list of features, and a development guide. This is typically a `README` file or a landing page on the web. The link to the documentation is set as `homepage_url` in the manifest. A great example is the [`README` of the GitHub plugin](https://github.com/mattermost/mattermost-plugin-github/blob/master/README.md). Typical components of documentation include:
```
Comment (aaronrothschild) [inline .github/ISSUE_TEMPLATE/add_plugin.md:62]: Can we put the development guide as a separate linked .MD file?  Of the hundreds of people who read it, only a small fraction will ever custom code or contribute.  and ittakes up a large section.  I'd be happy to put a "How to Contribute" link higher up in the doc more prominently if it was linked out.  Otherwise, I say sink it to the bottom of the readme.

**Review by aaronrothschild — APPROVED** [review]
LGTM

File: .github/ISSUE_TEMPLATE/add_plugin.md (hunk 7 above)
Comment (justinegeffen) [reply .github/ISSUE_TEMPLATE/add_plugin.md:62]: Looking at this, I agree with you. It's too wordy. I believe we have this documented elsewhere but if not it certainly can be moved. I'll address this in a separate PR. Thank you!
